src/Bonnier/Trapp/ServiceBase.php

```diff
@@ -0,0 +1,27 @@
+<?php
+namespace Bonnier\Trapp;
+
+use Bonnier\IndexSearch\ServiceRestBase;
+
+class ServiceBase extends ServiceRestBase {
+
+	protected $development;
+	protected $type;
+
+	public function __construct($username, $secret, $type) {
+		parent::__construct($username, $secret);
+		$this->type = $type;
+	}
+
+	// TODO: Add production/staging enviroment when ready (see above)
+
+	protected function getServiceUrl() {
+		return 'http://staging-trapp.whitealbum.dk/api/v1/' . $this->type;
```

Comment:
Vi kan godt lave det i ```ServiceBase``` (der findes en for både Trapp og IndexSearch) - men kun hvis API'et understøtter en forbindelse hvor man kan fravælge SSL.

Mht. til at smide den i ```HttpRequest``` så synes jeg ikke det giver mening.

setSSL - ville jo skifte protokol. Men en protokol kan være alt, det kunne også være steam:// eller skype:// - og det er egentlig noget man bør tage stilling til inden man opretter sin `HttpRequest`.

`HttpRequest` former egentlig bare det request til den url du vælger, så hvis den skulle kunne det - burde den tage imod et Url objekt som havde et ```setScheme`` eller noget i den duer.

Eksempel:

```$this->service->getUrl()->setScheme('https');````
```$this->service->getUrl()->setHost('lort.dk');```

Lad mig lige høre hvad du synes...